clang/lib/Sema/SemaTemplateInstantiateDecl.cpp

```diff
@@ -4952,7 +4952,8 @@ void Sema::InstantiateFunctionDefinition(SourceLocation PointOfInstantiation,
         std::make_pair(Function, PointOfInstantiation));
     } else if (TSK == TSK_ImplicitInstantiation) {
       if (AtEndOfTU && !getDiagnostics().hasErrorOccurred() &&
-          !getSourceManager().isInSystemHeader(PatternDecl->getBeginLoc())) {
+          !getSourceManager().isInSystemHeader(PatternDecl->getBeginLoc()) &&
+          !Function->isVirtualAsWritten()) {
```

Comment:
Should we also check that it is pure? AFAIK `isVirtualAsWritten` returns true for function marked explicitly virtual. There is also `FunctionDecl::isPure()` to check that it is pure.

Reply:
Updated.